### PR TITLE
feat(jose): add JWK support

### PR DESCRIPTION
**Pull request by npmccallum**

### CODE REVIEW

File: crates/byte/src/lib.rs [hunk 1]
```diff
@@ -1,34 +1,87 @@
...
-
-use serde::{de::Error as _, Deserialize, Serialize};
-
-pub trait Config {
-    const CONFIG: base64::Config;
+//! This crate provides a [`Bytes`] type which wraps most types that represent
+//! a contiguous array of bytes. It provides implementations for easy
+//! conversions to and from Base64 representations in string contexts.
+
+#![no_std]
```
Comment (rjzak) [inline crates/byte/src/lib.rs:8]: Why the change to `no_std`?

Comment (npmccallum) [reply crates/byte/src/lib.rs:8]: @rjzak Because this is a library. If you're writing a library that doesn't explicitly need `std`, then it is a good idea to make it `no_std`. This way other people can use it in `no_std` contexts.

File: crates/type/src/tag/entry.rs [hunk 2]
```diff
@@ -3,7 +3,7 @@
 
 use super::super::tree;
 
-use drawbridge_jose::jws::Jws;
+use drawbridge_jose::{jws::Jws, Typed};
```
Comment (rvolosatovs) [inline crates/type/src/tag/entry.rs:6]: nit; unmerge, see above

File: crates/type/src/tag/entry.rs [hunk 3]
```diff
@@ -21,7 +21,7 @@ use serde::{Deserialize, Serialize};
 #[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
 #[serde(untagged)]
 pub enum Entry {
-    Signed(Jws),
+    Signed(Box<Jws>),
```
Comment (rvolosatovs) [inline crates/type/src/tag/entry.rs:24]: Why is the box necessary here?

File: crates/app/src/tags.rs [hunk 4]
```diff
@@ -5,7 +5,7 @@ use crate::{assert_repo, RepoStore, TagStore};
 
 use std::sync::Arc;
 
-use drawbridge_jose::jws::Jws;
+use drawbridge_jose::{jws::Jws, Typed};
```
Comment (rvolosatovs) [inline crates/app/src/tags.rs:8]: nit; see above

**Comment by rvolosatovs** [comment]
Did not verify validity of implementation against the spec and trust you on that part

File: crates/byte/Cargo.toml [hunk 5]
```diff
@@ -5,8 +5,5 @@ authors = ["Profian Inc"]
 edition = "2021"
 
 [dependencies]
-# Internal dependencies
```
[the remark this answers is not in the document]
Comment (npmccallum) [reply crates/byte/Cargo.toml:8]: Correct.

File: crates/type/src/tag/entry.rs (hunk 3 above)
Comment (npmccallum) [reply crates/type/src/tag/entry.rs:24]: I knew you would ask that. :) Clippy complains because the size difference between the `Signed` and `Unsigned` variants is too large. The other option is to disable the warning.

File: crates/client/src/tag.rs [hunk 6]
```diff
@@ -5,7 +5,7 @@ use super::{Node, Repository};
 
 use std::error::Error;
 
-use drawbridge_jose::jws::Jws;
+use drawbridge_jose::{jws::Jws, Typed};
```
[the remark this answers is not in the document]
Comment (npmccallum) [reply crates/client/src/tag.rs]: Done.

File: crates/jose/Cargo.toml [hunk 7]
```diff
@@ -6,9 +6,12 @@ edition = "2021"
...
-drawbridge-byte = { path = "../byte" }
+drawbridge-byte = { path = "../byte", features = ["serde"] }
 
 # External dependencies
 base64 = { version = "0.13.0", default-features = false, features = ["std"] }
 serde = { version = "1.0.136", default-features = false, features = ["derive"] }
 serde_json = { version = "1.0.79", default-features = false, features = ["std"] }
+url = { version = "2.2.2", default-features = false, features = ["serde"] }
+zeroize = { version = "1.5.5", default-features = false, features = ["alloc"] }
+mediatype = { version = "0.19.2", features = ["serde"] }
```
[the remark this answers is not in the document]
Comment (npmccallum) [reply crates/jose/Cargo.toml]: Done.

File: crates/jose/src/lib.rs [hunk 8]
```diff
@@ -5,4 +5,21 @@
 #![forbid(unsafe_code, clippy::expect_used, clippy::panic)]
 
 pub mod b64;
+pub mod jwk;
 pub mod jws;
+
+use b64::Bytes;
+use serde::{Deserialize, Serialize};
+
+pub trait Typed {
```
[the remark this answers is not in the document]
Comment (npmccallum) [reply crates/jose/src/lib.rs]: Done.

File: crates/type/src/tag/entry.rs (hunk 2 above)
Comment (npmccallum) [reply crates/type/src/tag/entry.rs]: Done.

File: crates/app/src/tags.rs (hunk 4 above)
Comment (npmccallum) [reply crates/app/src/tags.rs]: Done.

File: crates/type/src/tag/entry.rs (hunk 3 above)
Comment (npmccallum) [reply crates/type/src/tag/entry.rs]: I decided to disable the lint. That is also self-documenting.

**Review by rvolosatovs — APPROVED** [review]

File: crates/byte/src/lib.rs (hunk 1 above)
Comment (npmccallum) [reply crates/byte/src/lib.rs:8]: Yes. I'll do that in the future.
